### PR TITLE
Hipp 870 request production access success

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: app/controllers/application/RequestProductionAccessEndJourneyController.scala [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    userAnswers.get(ProvideSupportingInformationPage) match {
+      case Some(information) => Right(information)
+      case None => Left(controllers.application.routes.ProvideSupportingInformationController.onPageLoad())
+    }
+  }
+
+  private def buildAccessRequest(application: Application, supportingInformation: String, requestedBy: String)(implicit request: DataRequest[AnyContent]): Future[Either[Result, AccessRequestRequest]] = {
+    applicationApiBuilder.build(application).map {
+      case Right(applicationApis) =>
+        val accessRequestApis = applicationApis.map(applicationApi => {
```
Comment (PaulCDurham) [inline app/controllers/application/RequestProductionAccessEndJourneyController.scala:121]: I think we need some filtering here. If we go through multiple loops of add API/add endpoints etc then we don't always want to add all endpoints of all APIs.

applicationApis.filter(_.endpoints.exists(_.primaryAccess == Inaccessible)).map(... etc etc

File: app/controllers/application/RequestProductionAccessEndJourneyController.scala [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          val accessRequestEndpoints = applicationApi.endpoints.map(endpoint => AccessRequestEndpoint(endpoint.httpMethod, endpoint.path, endpoint.scopes))
```
Comment (PaulCDurham) [inline app/controllers/application/RequestProductionAccessEndJourneyController.scala:122]: Similarly we only want inaccessible endpoints.

applicationApi.endpoints.filter(_.primaryAccess == Inaccessible).map(... etc

File: app/controllers/application/RequestProductionAccessEndJourneyController.scala (hunk 1 above)
Comment (llmikeyj) [reply app/controllers/application/RequestProductionAccessEndJourneyController.scala:121]: These apis are generated using the same code as for the request production access page; see 

RequestProductionAccessController
```

Is this also wrong?
```

File: app/views/application/RequestProductionAccessSuccessView.scala.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                value = Value(
+                    content = HtmlContent(app.name)
+                ),
+            ),
+            SummaryListRow(
+                key = Key(
+                    content = Text(messages("requestProductionAccessSuccess.apiName"))
+                ),
+                value = Value(
+                    content = Text(apis.head.apiName)
```
Comment (PaulCDurham) [inline app/views/application/RequestProductionAccessSuccessView.scala.html]: Could you have a chat with Jonathon about adding multiple APIs? Perhaps in that scenario we should display all the API names. Perhaps add that as a UI improvement ticket?

**Review by PaulCDurham — CHANGES_REQUESTED** [review]
One thing to fix, one question for the review with Jonathon

File: app/views/application/RequestProductionAccessSuccessView.scala.html (hunk 3 above)
Comment (PaulCDurham) [reply app/views/application/RequestProductionAccessSuccessView.scala.html]: I see you've done that, cool

**Review by PaulCDurham — APPROVED** [review]

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**